R/validate-config-utils.R

```diff
@@ -535,28 +535,64 @@ validate_round_derived_task_ids <- function(round, round_i, schema) {
   }
   task_ids <- get_round_task_ids(round)
   invalid_derived_task_ids <- setdiff(derived_task_ids, task_ids)
-  if (length(invalid_derived_task_ids) == 0L) {
+  valid_derived_task_ids <- setdiff(
+    derived_task_ids,
+    invalid_derived_task_ids
+  )
+  req_derived_task_ids <- derived_task_ids_with_required_vals(
+    round, valid_derived_task_ids,
+    depth = "round"
+  )
+  names_valid <- length(invalid_derived_task_ids) == 0L
+  no_req_values <- length(req_derived_task_ids) == 0L
+
+  if (names_valid && no_req_values) {
     return(NULL)
   }
-
-  tibble::tibble(
-    instancePath = glue::glue_data(
-      list(round_i = round_i),
-      get_error_path(
-        schema,
-        "rounds/items/properties/derived_task_ids",
-        "instance"
+  out <- NULL
+  if (!names_valid) {
+    out <- tibble::tibble(
+      instancePath = glue::glue_data(
+        list(round_i = round_i),
+        get_error_path(
+          schema,
+          "rounds/items/properties/derived_task_ids",
+          "instance"
+        )
+      ),
+      schemaPath = get_error_path(
+        schema, "rounds/items/properties/derived_task_ids",
+        "schema"
+      ),
+      keyword = "round derived task IDs",
+      message = "derived_task_ids values MUST MATCH valid round task_id variable names",
+      schema = paste(task_ids, collapse = ", "),
+      data = paste(invalid_derived_task_ids, collapse = ", ")
+    )
+  }
+  if (!no_req_values) {
+    out <- rbind(
```

Comment:
Should this be binding the previous `out`?

```suggestion
    out <- rbind(
             out,
```

Reply:
I think this isn't being caught by the tests because the round-level `derived_task_ids` for `v4-tasks-fail-dynamic.json` does not have an unknown task id listed.

Reply:
Good catch! Fixed and boulstered the test

Reply:
https://github.com/hubverse-org/hubAdmin/pull/82/commits/c730d0746093244c9088a32978edaee13e986057